core/src/main/java/org/openstack4j/openstack/workflow/internal/WorkbookDefinitionServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package org.openstack4j.openstack.workflow.internal;
+
+import org.openstack4j.api.workflow.WorkbookDefinitionService;
+import org.openstack4j.core.transport.HttpResponse;
+import org.openstack4j.model.common.payloads.InputStreamPayload;
+import org.openstack4j.model.workflow.Scope;
+import org.openstack4j.model.workflow.WorkbookDefinition;
+import org.openstack4j.openstack.workflow.domain.MistralWorkbookDefinition;
+import org.openstack4j.openstack.workflow.domain.MistralWorkbookDefinition.MistralWorkbookDefinitions;
+
+import java.io.InputStream;
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Workbook definition service implementation.
+ *
+ * @author Renat Akhmerov
+ */
+public class WorkbookDefinitionServiceImpl extends BaseMistralService implements WorkbookDefinitionService {
+
+    @Override
+    public List<? extends WorkbookDefinition> list() {
+        return get(MistralWorkbookDefinitions.class, uri("/workbooks")).execute().getList();
+    }
+
+    @Override
+    public WorkbookDefinition create(InputStream wfText, Scope scope) {
+        checkNotNull(wfText);
+        checkNotNull(scope);
+
+        Invocation<MistralWorkbookDefinition> invocation = post(
+                MistralWorkbookDefinition.class,
+                uri("/workbooks?scope=%s", scope.toString().toLowerCase())
+        );
+
+        return invocation.entity(new InputStreamPayload(wfText)).execute();
+    }
+
+    @Override
+    public WorkbookDefinition get(String identifier) {
+        return get(MistralWorkbookDefinition.class, uri("/workbooks/%s", identifier)).execute();
+    }
+
+    @Override
+    public HttpResponse delete(String identifier) {
```

Comment:
Ok, let me recall.. I used HTTPResponse because ActionResponse masks a real HTTP code. No matter what I configured with "respondWith()" (e.g. 204) in tests it always returned 200. This is where it happens: https://github.com/ContainX/openstack4j/blob/master/core/src/main/java/org/openstack4j/core/transport/HttpEntityHandler.java#L63
Above that line there's a condition "if (response.getStatus() >= 400) {". So id status is less than 400 then it always returns "ActionResponse.actionSuccess()" which is 200.

On the other hand, precise 204 code returned by Mistral endpoint for "delete" may not be so important in the Java client.

What do you think? Maybe I don't fully understand semantics of ActionResponse.

Any help would be useful.

Thanks